airflow/example_dags/example_kubernetes_executor.py

```diff
@@ -16,88 +16,214 @@
-    def four_task():
+    @task(executor_config=executor_config_other_ns)
+    def other_namespace_task():
         print_stuff()
 
-    start_task = start_task()
-    one_task = one_task()
-    two_task = assert_zip_binary()
-    three_task = three_task()
-    four_task = four_task()
+    other_ns_task = other_namespace_task()
+
+    # You can also change the base image, here we used the worker image for demonstration.
+    # Note that the image must have the same configuration as the
+    # worker image. Could be that you want to run this task in a special docker image that has a zip
+    # library built-in. You build the special docker image on top your worker image.
+    kube_exec_config_special = {
+        "pod_override": k8s.V1Pod(
+            spec=k8s.V1PodSpec(
+                containers=[
+                    k8s.V1Container(
+                        name="base", image=f"{worker_container_repository}:{worker_container_tag}"
+                    ),
+                ]
+            )
+        )
+    }
+
+    @task(executor_config=kube_exec_config_special)
+    def base_image_override_task():
+        print_stuff()
+
+    base_image_task = base_image_override_task()
+
+    # Use k8s_client.V1Affinity to define node affinity
+    k8s_affinity = k8s.V1Affinity(
+        pod_anti_affinity=k8s.V1PodAntiAffinity(
+            required_during_scheduling_ignored_during_execution=[
+                k8s.V1PodAffinityTerm(
+                    label_selector=k8s.V1LabelSelector(
+                        match_expressions=[
+                            k8s.V1LabelSelectorRequirement(key='app', operator='In', values=['airflow'])
+                        ]
+                    ),
+                    topology_key='kubernetes.io/hostname',
+                )
+            ]
+        )
+    )
+
+    # Use k8s_client.V1Toleration to define node tolerations
+    k8s_tolerations = [k8s.V1Toleration(key='dedicated', operator='Equal', value='airflow')]
+
+    # Use k8s_client.V1ResourceRequirements to define resource limits
+    k8s_resource_requirements = k8s.V1ResourceRequirements(
+        requests={'memory': '512Mi'}, limits={'memory': '512Mi'}
+    )
+
+    kube_exec_config_resource_limits = {
+        "pod_override": k8s.V1Pod(
+            spec=k8s.V1PodSpec(
+                containers=[
+                    k8s.V1Container(
+                        name="base",
+                        resources=k8s_resource_requirements,
+                    )
+                ],
+                affinity=k8s_affinity,
+                tolerations=k8s_tolerations,
+            )
+        )
+    }
+
+    @task(executor_config=kube_exec_config_resource_limits)
+    def task_with_resource_limits():
+        print_stuff()
+
+    four_task = task_with_resource_limits()
 
-    start_task >> [one_task, two_task, three_task, four_task]
+    (start_task >> [volume_task, other_ns_task, sidecar_task] >> third_task >> [base_image_task, four_task])
```

Comment:
```suggestion
    start_task >> [volume_task, other_ns_task, sidecar_task] >> third_task >> [base_image_task, four_task]
```

Are the outmost parantheses needed? (nit)

Reply:
Not needed, I previously had a lot of tasks here